modules/metastorage-client/src/integrationTest/java/org/apache/ignite/internal/metastorage/client/ITMetaStorageServiceTest.java

```diff
@@ -791,112 +606,88 @@ public void testRangeClose() throws Exception {
-                                }
-
-                                @Override public org.apache.ignite.internal.metastorage.server.WatchEvent next() {
-                                    return returnedWatchEvents;
-                                }
-                            };
-
-                            AtomicInteger retirevedItemCnt = new AtomicInteger(0);
-
-                            @Override public void close() throws Exception {
-                                // No-op.
-                            }
+        when(mockStorage.watch(keyFrom.bytes(), keyTo.bytes(), rev)).thenAnswer(invocation -> {
+            var cursor = mock(Cursor.class);
 
-                            @NotNull @Override public Iterator<org.apache.ignite.internal.metastorage.server.WatchEvent> iterator() {
-                                return it;
-                            }
+            when(cursor.hasNext()).thenReturn(true);
+            when(cursor.next()).thenReturn(expectedEvent);
 
-                            @Override public boolean hasNext() {
-                                return it.hasNext();
-                            }
-
-                            @Override
-                            public org.apache.ignite.internal.metastorage.server.WatchEvent next() {
-                                return it.next();
-                            }
-                        };
-                    }
-                });
+            return cursor;
+        });
 
         CountDownLatch latch = new CountDownLatch(1);
 
         IgniteUuid watchId = metaStorageSvc.watch(keyFrom, keyTo, rev, new WatchListener() {
-            @Override public boolean onUpdate(@NotNull WatchEvent events) {
-                List gotEvents = new ArrayList();
-
-                List returnedWatchEvents = new ArrayList(events.entryEvents());
-
-                Iterator<EntryEvent> iter = events.entryEvents().iterator();
+            @Override public boolean onUpdate(@NotNull WatchEvent event) {
+                Collection<EntryEvent> expectedEvents = expectedEvent.entryEvents();
+                Collection<org.apache.ignite.internal.metastorage.client.EntryEvent> actualEvents = event.entryEvents();
 
-                while (iter.hasNext())
-                    gotEvents.add(iter.next());
+                assertEquals(expectedEvents.size(), actualEvents.size());
 
-                assertEquals(3, gotEvents.size());
+                Iterator<EntryEvent> expectedIterator = expectedEvents.iterator();
+                Iterator<org.apache.ignite.internal.metastorage.client.EntryEvent> actualIterator = actualEvents.iterator();
 
-                assertTrue(gotEvents.contains(returnedWatchEvents.get(0)));
+                while (expectedIterator.hasNext() && actualIterator.hasNext()) {
+                    org.apache.ignite.internal.metastorage.server.EntryEvent expectedEntryEvent = expectedIterator.next();
```

Comment:
What's up with all these full names? Are there collisions somewhere? I don't get it

Reply:
yes, there are `metastorage.server` and `metastorage.client` packages which contain classes with the same name but slightly different structures =)